### PR TITLE
[ESP32] Add ethernet support for commissioning 

### LINKED ISSUE #22511: [Feature] Ethernet Commissioning using esp32-ethernet-kit
### Feature description

Support of ethernet for commissioning using esp32

### Platform

esp32

### PR DESCRIPTION
**Pull request by PSONALl**
#### Problem
Added support of ethernet in all-clusters-app/esp32
Fixes #22511 
#### Change overview
- Added initialisation and event handlers for ethernet in case of esp32

#### Testing
- Tested commissioning and cluster control

### CODE REVIEW

**Comment by andy31415** [comment]
Accepted to 1.0 as a platform specific fix.

@PSONALl - needs restyle

**Comment by dhrishi** [comment]
The changes related to config options are yet to be done. Moving it to draft until that happens

**Comment by PSONALl** [comment]
> The changes related to config options are yet to be done. Moving it to draft until that happens

Done with the changes

**Review by andy31415 — APPROVED** [review]

File: config/esp32/components/chip/Kconfig [hunk 1]
```diff
@@ -516,6 +516,51 @@ menu "CHIP Device Layer"
 
     endmenu
 
+    menu "Ethernet Options"
+        visible if ENABLE_ETHERNET_TELEMETRY
+
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:521]: help here depending on what type of ethernet is supported

File: config/esp32/components/chip/Kconfig [hunk 2]
```diff
@@ -516,6 +516,51 @@ menu "CHIP Device Layer"
...
+            int 
+            default 33 if IDF_TARGET_ESP32
+            default 46 if IDF_TARGET_ESP32S2
+            default 19 if IDF_TARGET_ESP32C3
+            default 48 if IDF_TARGET_ESP32S3
+
+        config ETH_MDC_GPIO
+            int "SMI MDC GPIO number"
+            range GPIO_RANGE_MIN GPIO_RANGE_MAX
+            default 23 if ENABLE_ETHERNET_TELEMETRY
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:536]: Why is this `if ENABLE_ETHERNET_TELEMETRY` required here and at other places here?

File: docs/guides/esp32/build_app_and_commission.md [hunk 3]
```diff
@@ -190,6 +191,16 @@ $ out/debug/chip-tool pairing ble-wifi 12345 MY_SSID MY_PASSWORD 20202021 3840
      $ ./out/debug/chip-tool pairing ble-thread 12345 hex:<operational-dataset> 20202021 3840
     ```
 
+#### Commissioning the Ethernet device (ESP32-Ethernet-Kit)
+
+```
+$ out/debug/chip-tool pairing ethernet 12345 20202021 3840 device-remote-ip device-remote-port
+```
+
+note: In order to commission an ethernet device, from all-clusters-app enable
```
Comment (dhrishi) [inline docs/guides/esp32/build_app_and_commission.md:200]: Nit:
note: => Note:

File: docs/guides/esp32/build_app_and_commission.md [hunk 4]
```diff
@@ -190,6 +191,16 @@ $ out/debug/chip-tool pairing ble-wifi 12345 MY_SSID MY_PASSWORD 20202021 3840
      $ ./out/debug/chip-tool pairing ble-thread 12345 hex:<operational-dataset> 20202021 3840
     ```
 
+#### Commissioning the Ethernet device (ESP32-Ethernet-Kit)
+
+```
+$ out/debug/chip-tool pairing ethernet 12345 20202021 3840 device-remote-ip device-remote-port
```
Comment (dhrishi) [inline docs/guides/esp32/build_app_and_commission.md:197]: Replace device-remote-port by the actual value?

File: config/esp32/components/chip/Kconfig (hunk 2 above)
Comment (PSONALl) [reply config/esp32/components/chip/Kconfig:536]: If not included .. these config options will be generated in other cases too when the option is not enabled

File: docs/guides/esp32/build_app_and_commission.md (hunk 4 above)
Comment (PSONALl) [reply docs/guides/esp32/build_app_and_commission.md:197]: ok

File: config/esp32/components/chip/Kconfig (hunk 2 above)
Comment (dhrishi) [reply config/esp32/components/chip/Kconfig:536]: But the usage of these config options in the code is already guarded under the ethernet flag, no?

Comment (PSONALl) [reply config/esp32/components/chip/Kconfig:536]: Yes they will not be used but still they will be generated in sdkconfig

Comment (dhrishi) [reply config/esp32/components/chip/Kconfig:536]: Ok. So, in default cases none of the ethernet options are seen in the sdkconfig. You've confirmed that, right?

Comment (PSONALl) [reply config/esp32/components/chip/Kconfig:536]: yes

File: config/esp32/components/chip/Kconfig (hunk 1 above)
Comment (PSONALl) [reply config/esp32/components/chip/Kconfig:521]: I added help under ENABLE_ETHERNET_TELEMETRY option

**Review by dhrishi — APPROVED** [review]

**Pull request merged**